fhir-model/src/test/java/com/ibm/fhir/model/test/ModelBuilderTest.java

```diff
@@ -0,0 +1,91 @@
+import com.ibm.fhir.model.type.code.PublicationStatus;
+
+public class ModelBuilderTest {
+    @Test
+    public void testPrimitiveSetterEquivalence() {
+        Patient p1 = Patient.builder()
+                .meta(Meta.builder()
+                    .lastUpdated(ZonedDateTime.of(2021, 8, 19, 00, 59, 59, 0, ZoneOffset.of("-05:00"))) // Instant
+                    .build())
+                .contained(ValueSet.builder()
+                    .status(PublicationStatus.DRAFT)
+                    .expansion(Expansion.builder()
+                        .timestamp(DateTime.of("2021-08-19T00:59:59-05:00"))
+                        .total(0)                                                                       // Integer
+                        .build())
+                    .build())
+                .active(false)                                                                          // Boolean
+                .birthDate(LocalDate.of(1984, 9, 4))                                                    // Date
+//                .multipleBirth(1)     // should we add overloads for choice types that can take FHIR String, Boolean, Integer, Date, or Instant?
+                .name(HumanName.builder()
+                    .given("Lee")                                                                       // String
+                    .build())
+                .build();
+
+        Patient p2 = Patient.builder()
+                .meta(Meta.builder()
+                    .lastUpdated(com.ibm.fhir.model.type.Instant.of("2021-08-19T00:59:59-05:00"))       // Instant
+                    .build())
+                .contained(ValueSet.builder()
+                    .status(PublicationStatus.DRAFT)
+                    .expansion(Expansion.builder()
+                        .timestamp(DateTime.of("2021-08-19T00:59:59-05:00"))
+                        .total(com.ibm.fhir.model.type.Integer.of(0))                                   // Integer
+                        .build())
+                    .build())
+                .active(com.ibm.fhir.model.type.Boolean.FALSE)                                          // Boolean
+                .birthDate(com.ibm.fhir.model.type.Date.of("1984-09-04"))                               // Date
+//                .multipleBirth(com.ibm.fhir.model.type.Integer.of(1))
+                .name(HumanName.builder()
+                    .given(string("Lee"))                                                               // String
+                    .build())
+                .build();
+
+        assertEquals(p1, p2);
+    }
+
+    @Test(expectedExceptions = IllegalStateException.class)
+    public void testVarArgWithNull() {
+        Basic.builder()
+            .code(CodeableConcept.builder()
+                .coding((Coding)null)
+                .build())
+            .build();
+    }
+
+    // This one throws a NullPointerException rather than the expected IllegalStateException...is that an issue?
```

Comment:
is this an issue?

Reply:
personally I think its fine to throw the NPE in this case, but maybe we could add it to the javadoc